File: api/src/functions/health.ts
import { app, HttpResponseInit } from '@azure/functions'
import { initDatabase } from '../cosmosdb.js'

async function healthHandler(): Promise<HttpResponseInit> {
  try {
    await initDatabase()
    return { jsonBody: { status: 'healthy' } }
  } catch {
    return { status: 503, jsonBody: { status: 'unhealthy' } }
  }
}

app.http('health', {
  methods: ['GET'],
  authLevel: 'anonymous',
  route: 'health',
  handler: healthHandler,
})

File: api/app/routers/health.py
from fastapi import APIRouter
from fastapi.responses import JSONResponse

from app.cosmosdb import check_database_connection

router = APIRouter()


@router.get("/health")
async def health():
    try:
        await check_database_connection()
        return {"status": "healthy"}
    except Exception:
        return JSONResponse({"status": "unhealthy"}, status_code=503)
